R/CallbackSetTB.R

```diff
@@ -0,0 +1,75 @@
+#' @title TensorBoard Logging Callback
+#'
+#' @name mlr_callback_set.tb
+#'
+#' @description
+#' Logs training loss and validation measures as events that can be tracked using TensorBoard.
+#' @details
+#' Logs at most every epoch.
+#'
+#' @param path (`character(1)`)\cr
+#'   The path to a folder where the events are logged.
+#'   Point TensorBoard to this folder to view them.
+#' @family Callback
+#' @export
+#' @include CallbackSet.R
+CallbackSetTB = R6Class("CallbackSetTB",
+  inherit = CallbackSet,
+  lock_objects = FALSE,
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function(path) {
+      self$path = assert_path_for_output(path)
+      if (!dir.exists(path)) {
+        dir.create(path, recursive = TRUE)
+      }
+    },
+    #' @description
+    #' Logs the training loss and validation measures as TensorFlow events.
+    #' Meaningful changes happen at the end of each epoch.
+    #' Notably NOT on_batch_valid_end, since there are no gradient steps between validation batches,
+    #' and therefore differences are due to randomness
+    # TODO: display the appropriate x axis with its label in TensorBoard
```

Comment:
Use the step argument in log_event calls

Reply:
What's ambiguous here?

Reply:
I only did some basic sanity checking (basically just looked at the test case in TensorBoard), so wasn't 100% confident in the correctness at the time of writing. But I think there's not a whole lot to mess up here